packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.js

```diff
@@ -1,89 +1,52 @@
-import React, { Children, cloneElement } from 'react';
-import styles from '@patternfly/patternfly/components/AppLauncher/app-launcher.css';
-import { css } from '@patternfly/react-styles';
+import React from 'react';
 import PropTypes from 'prop-types';
-import ApplicationLauncherMenu from './ApplicationLauncherMenu';
-import Toggle from './Toggle';
+import { Dropdown, DropdownDirection, DropdownPosition, DropdownToggle } from '../Dropdown';
 import { ThIcon } from '@patternfly/react-icons';
-import { DropdownContext } from '../Dropdown/dropdownConstants';
-import GenerateId from '../../helpers/GenerateId/GenerateId';
 
 export const propTypes = {
   /** Additional element css classes */
   className: PropTypes.string,
+  /** Display menu above or below dropdown toggle */
+  direction: PropTypes.oneOf(Object.values(DropdownDirection)),
   /** Array of DropdownItem nodes that will be rendered in the dropdown Menu list */
   dropdownItems: PropTypes.array,
+  /** Render Application launcher toggle as disabled icon */
+  isDisabled: PropTypes.bool,
   /** open bool */
   isOpen: PropTypes.bool,
   /** Function callback called when user selects item */
   onSelect: PropTypes.func,
   /** Callback called when application launcher toggle is clicked */
   onToggle: PropTypes.func,
+  /** Indicates where menu will be alligned horizontally */
+  position: PropTypes.oneOf(Object.values(DropdownPosition)),
   /** Adds accessible text to the button. Required for plain buttons */
   'aria-label': PropTypes.string
 };
 
 export const defaultProps = {
   className: '',
+  isDisabled: false,
+  direction: DropdownDirection.down,
   dropdownItems: [],
   isOpen: false,
   onSelect: Function.prototype,
   onToggle: Function.prototype,
+  position: DropdownPosition.left,
   'aria-label': 'Actions'
 };
 
-class ApplicationLauncher extends React.Component {
-  onEnter = () => {
-    this.openedOnEnter = true;
-  };
-
-  componentDidUpdate() {
-    if (!this.props.isOpen) this.openedOnEnter = false;
-  }
-
-  render() {
-    const { 'aria-label': ariaLabel, dropdownItems, className, isOpen, onSelect, onToggle } = this.props;
-    return (
-      <GenerateId>
-        {randomId => (
-          <div
-            className={css(styles.appLauncher, isOpen && styles.modifiers.expanded, className)}
```

Comment:
What's the practical difference between `pf-c-app-launcher` and `pf-c-dropdown` @mcoker ? The [current AppLauncher](http://patternfly-react.surge.sh/patternfly-4/components/applicationlauncher/) and [this Dropdown AppLauncher](https://2101-pr-patternfly-react-patternfly.surge.sh/patternfly-4/components/applicationlauncher/) look the same.

Reply:
@redallen sorry I missed this. They're effectively the same currently. The app launcher is a dropdown with `.pf-m-plain` on the `.pf-c-dropdown__toggle` element, just using a different icon than we have in our examples.

What the dropdown doesn't allow for is a future variation of the application launcher where we re-arrange the items in the dropdown to be square or whatever (like this - https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/application-launcher.html)